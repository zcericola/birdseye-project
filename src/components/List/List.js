import React, { useState } from "react";
import Grid from "@material-ui/core/Grid";
import useStyles from "./listStyles";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import TabContent from "../TabContent/TabContent";
import ListItem from "../ListItem/ListItem";

const List = ({ listTitle, showTabs, date, exTasks }) => {
  const {
    title,
    taskTab,
    taskCreator,
    dateSubtitle,
    completedTitle,
    dashedLines,
    completedSection
  } = useStyles();
  const [selectedTab, setSelectedTab] = useState(0);
  const [list, addNewTask] = useState();

  const handleTabSelection = (e, selectedTab) => {
    setSelectedTab(selectedTab);
  };

  const createTask = name => {
    //logic to add a new task would go here
    // addNewTask( { name } );
  };

  return (
    <Grid item>
      <div className={title}>
        <Typography variant={"h5"}>{listTitle}</Typography>
        <Typography variant={"h6"} className={dateSubtitle}>
          {date && date}
        </Typography>
      </div>
      {showTabs && (
        <Tabs value={selectedTab} onChange={handleTabSelection}>
          <Tab label="All" className={taskTab} />
          <Tab className={taskTab} label="Projects" />
        </Tabs>
      )}
      {selectedTab === 0 && (
        <Grid item>
          <TabContent>
            <Button component="div" className={taskCreator} onClick={createTask}>
              +Create Task
                        </Button>
            {exTasks &&
              exTasks.map((task, idx) => {
                return <ListItem key={idx} taskName={task.name} scheduledAt={task.scheduledAt} />;
              })}
          </TabContent>
          {listTitle === "Today" && (
            <section className={completedSection}>
              <h2 className={dashedLines}></h2>
              <span className={completedTitle}>Completed</span>
              <h2 className={dashedLines}></h2>
            </section>
          )}
        </Grid>
      )}

      {selectedTab === 1 && <TabContent>Projects would go here...</TabContent>}
    </Grid>
  );
};

export default List;
