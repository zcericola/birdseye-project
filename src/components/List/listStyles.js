/* eslint-disable import/no-extraneous-dependencies */
import { makeStyles } from '@material-ui/styles';
import {
    primaryColor
    , purpleGray
    , centerVandH
    , secondaryColor
    , textGray
    , accentGray
} from '../../globalStyles';
import { relative } from 'path';

const useStyles = makeStyles( {
    ListContainer: {
        backgroundColor: '#fff'
    }
    , title: {
        margin: '10px 30px'
        , color: primaryColor
    }

    , taskCreator: {
        ...centerVandH
        , height: 100
        , color: secondaryColor
        , backgroundColor: purpleGray
        , textTransform: 'uppercase'
        , cursor: 'pointer'
        , margin: 0
    }
    , dateSubtitle: {
        color: textGray
        , fontSize: 12
    }
    , dashedLines: {
        width: '40%'
        , borderBottom: '1px dashed black'
        , lineHeight: 1
        , display: 'inline-block'
        , color: textGray
        , margin: 'inherit auto'
    }
    , completedTitle: {
        color: textGray
        , margin: '0 20px'
        , lineHeight: '10px'
        , position: 'relative'
        , bottom: 15
        , fontStyle: 'italic'

    }
    , completedSection: {
        margin: '50px auto'
        , width: '80%'
    }

} );

export default useStyles;
