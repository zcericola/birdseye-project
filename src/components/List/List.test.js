import React from 'react';
import { mount, debug } from 'enzyme';
import List from './List';
import Typography from '@material-ui/core/Typography';
import renderer from 'react-test-renderer';
import '../../setup';

describe('List component', () => {

    it('matches the snapshot', () => {
        const tree = renderer.create(<List />).toJSON();
        expect(tree).toMatchSnapshot();

    });
    it('title should have two inner elements', () => {
        const wrapper = mount(<List />);
        const title = wrapper.find(Typography);
        expect(title.length).toEqual(2);
        wrapper.unmount();
    });
});
