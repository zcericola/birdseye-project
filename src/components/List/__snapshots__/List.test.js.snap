// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component matches the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <div
    className="makeStyles-title-2"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    />
    <h6
      className="MuiTypography-root makeStyles-dateSubtitle-4 MuiTypography-h6"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiGrid-root makeStyles-contentContainer-141 MuiGrid-item"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root makeStyles-taskCreator-3 MuiButton-text"
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnd={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          +Create Task
        </span>
      </div>
    </div>
  </div>
</div>
`;
